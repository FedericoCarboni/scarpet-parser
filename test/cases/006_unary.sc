// Unary Operator precedence test
!id:attr && true