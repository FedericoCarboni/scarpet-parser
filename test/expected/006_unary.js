/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: '&&',
    lvalue: {
        kind: 'UnaryExpression',
        operator: '!',
        value: {
            kind: 'BinaryExpression',
            operator: ':',
            lvalue: {
                kind: 'Variable',
                name: 'id',
                start: {
                    offset: 35,
                    line: 1,
                    character: 1,
                },
                end: {
                    offset: 37,
                    line: 1,
                    character: 3,
                },
            },
            rvalue: {
                kind: 'Variable',
                name: 'attr',
                start: {
                    offset: 38,
                    line: 1,
                    character: 4,
                },
                end: {
                    offset: 43,
                    line: 1,
                    character: 9,
                },
            },
            comment: '// Unary Operator precedence test\n',
            start: {
                offset: 35,
                line: 1,
                character: 1,
            },
            end: {
                offset: 43,
                line: 1,
                character: 9,
            },
        },
        start: {
            offset: 34,
            line: 1,
            character: 0,
        },
        end: {
            offset: 43,
            line: 1,
            character: 9,
        },
    },
    rvalue: {
        kind: 'Constant',
        name: 'true',
        start: {
            offset: 46,
            line: 1,
            character: 12,
        },
        end: {
            offset: 50,
            line: 1,
            character: 16,
        },
    },
    comment: '// Unary Operator precedence test\n',
    start: {
        offset: 34,
        line: 1,
        character: 0,
    },
    end: {
        offset: 50,
        line: 1,
        character: 16,
    },
};
