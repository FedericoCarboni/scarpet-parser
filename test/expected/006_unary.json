{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": "&&",
    "lvalue": {
      "kind": "UnaryExpression",
      "operator": "!",
      "value": {
        "kind": "BinaryExpression",
        "operator": ":",
        "lvalue": {
          "kind": "Variable",
          "name": "id",
          "start": {
            "pos": 35,
            "row": 1,
            "col": 1
          },
          "end": {
            "pos": 37,
            "row": 1,
            "col": 3
          }
        },
        "rvalue": {
          "kind": "Variable",
          "name": "attr",
          "start": {
            "pos": 38,
            "row": 1,
            "col": 4
          },
          "end": {
            "pos": 43,
            "row": 1,
            "col": 9
          }
        },
        "comment": "// Unary Operator precedence test\n",
        "start": {
          "pos": 35,
          "row": 1,
          "col": 1
        },
        "end": {
          "pos": 43,
          "row": 1,
          "col": 9
        }
      },
      "start": {
        "pos": 34,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 43,
        "row": 1,
        "col": 9
      }
    },
    "rvalue": {
      "kind": "Constant",
      "name": "true",
      "start": {
        "pos": 46,
        "row": 1,
        "col": 12
      },
      "end": {
        "pos": 50,
        "row": 1,
        "col": 16
      }
    },
    "comment": "// Unary Operator precedence test\n",
    "start": {
      "pos": 34,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 50,
      "row": 1,
      "col": 16
    }
  }
}